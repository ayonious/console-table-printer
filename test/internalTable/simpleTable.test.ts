import {
  renderSimpleTable,
  printSimpleTable,
} from '../../src/internalTable/internal-table-printer';

describe('Example: Print a simple Table without table instance creation', () => {
  it('simple print', () => {
    const testCases = [
      { index: 3, text: 'I would like some gelb bananen bitte', value: 100 },
      { index: 4, text: 'I hope batch update is working', value: 300 },
    ];
    const tableStr: string = renderSimpleTable(testCases);

    expect(tableStr).toMatchSnapshot();
  });

  it('Make sure this function works dont throw exception', () => {
    const testCases = [
      { index: 3, text: 'I would like some gelb bananen bitte', value: 100 },
      { index: 4, text: 'I hope batch update is working', value: 300 },
    ];
    printSimpleTable(testCases);
  });
});
