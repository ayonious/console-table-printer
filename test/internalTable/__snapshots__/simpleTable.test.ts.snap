// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table without table instance creation simple print 1`] = `
"┌───────┬──────────────────────────────────────┬───────┐
│[37m [0m[01mindex[0m │[37m [0m[01m                                text[0m │[37m [0m[01mvalue[0m │
├───────┼──────────────────────────────────────┼───────┤
│[37m [0m[37m    3[0m │[37m [0m[37mI would like some gelb bananen bitte[0m │[37m [0m[37m  100[0m │
│[37m [0m[37m    4[0m │[37m [0m[37m      I hope batch update is working[0m │[37m [0m[37m  300[0m │
└───────┴──────────────────────────────────────┴───────┘"
`;
